### PR TITLE
User: Hook up user token saving to Redux store, fix typo, stub Device settings

### LINKED ISSUE #71: Map: Add recenter to group's default coordinates
Commit in this PR: https://github.com/Cosecha/redadalertas/pull/70

### PR DESCRIPTION
**Pull request by arghgr**
Update to #62 and closes #71 . All components that need to check when a user exists on the device can now do so by checking the Redux store.
- Upon mounting the EventsMap component, checkForUserLogin attempts to fetch user info from async storage on the device and makes sure it's valid, then saves the user token in the Redux store with saveUserToken.
- In ReporterLoginForm, a login is attempted (upon success, authServices.login saves the returned user info to the async store); if the login is successful and saved to the async store, the user info will be saved to the Redux store.
- In SettingsPage, after logout (which deletes user info from async storage), deleteUserToken() is called to remove user info from the Redux store.
- In ChangePassword, a login is attempted with user info fetched from async storage using checkForUserLogin. If the login is successful, the user's password is updated, and another login is attempted. If that works, then the user token is saved to the Redux store.

### CODE REVIEW

File: src/components/EventForm.js [hunk 1]
```diff
@@ -78,21 +79,18 @@ export default class EventForm extends Component {
...
+
+      let data = { ...values }
 
       if (this.props.newEvent === true) {
         response = await eventServices.post(data);
       } else {
         // If updating event, add event _id, fix/remove server-generated data
         if (data.created) delete data["created"];
+        if (data.updated) delete data["updated"];
+        // If updating event, add event _id
```
Comment (alexgraffeocohen) [inline src/components/EventForm.js:93]: I don't think we need this extra comment here.

File: src/components/EventForm.js [hunk 2]
```diff
@@ -266,3 +264,15 @@ export default class EventForm extends Component {
     );
   }
 }
+
+const mapStateToProps = state => ({
+  user: state.user
+});
+
+const mapDispatchToProps = dispatch => ({
+});
```
Comment (alexgraffeocohen) [inline src/components/EventForm.js:273]: Looks like we don't need to include mapDispatchToProps if we're not going to use it?

**Review by alexgraffeocohen — APPROVED** [review]
Great work!! The code got a lot simpler and more uniform, and there are some nice added features in here. A couple comments, but they're so minor that I'm approving anyways.

File: src/components/EventForm.js (hunk 2 above)
Comment (arghgr) [reply src/components/EventForm.js:273]: You're right, taking this out once I make sure it's not needed.

File: src/components/EventForm.js (hunk 1 above)
Comment (arghgr) [reply src/components/EventForm.js:93]: Fixing this as well-- the id key's underscore is no longer removed on the server side anyway.

**Pull request merged**